plugins/inputs/snmp/snmp.go

```diff
@@ -948,7 +948,7 @@ var snmpTranslateCachesLock sync.Mutex
 var snmpTranslateCaches map[string]snmpTranslateCache
 
 // snmpTranslate resolves the given OID.
-func snmpTranslate(oid string) (mibName string, oidNum string, oidText string, conversion string, err error) {
+func SnmpTranslate(oid string) (mibName string, oidNum string, oidText string, conversion string, err error) {
```

Comment:
It seems natural that snmp_trap would depend on snmp.  I don't see any benefit to moving the code to a common dir.

I generally don't like the idea of common dirs. It seems like they always hold stuff that would fit better elsewhere.

Reply:
Perhaps, we have some things that are either shared by many plugins, mostly these are in `internal` right now, which ought to be moved outside of this special directory so that they may be used by plugins outside of the repo.  I would be inclined to agree it doesn't make sense to have these in a common directory, but I don't want too many top level directories.  We also have some functions that are used from both, for example the kafka input and the kafka output where it's less clear which plugin should own the logic.